engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -2622,8 +2623,101 @@ namespace dmGameObject
             dynamic_update_context.m_AccumFrameTime = collection->m_FixedAccumTime;
         }
 
+        uint32_t num_fixed_steps = 0;
+        UpdateContext fixed_update_context;
+        if (update_context->m_FixedUpdateFrequency != 0 && update_context->m_TimeScale > 0.001f)
+        {
+            if (collection->m_FirstUpdate)
+            {
+                collection->m_FirstUpdate = 0;
+                collection->m_FixedAccumTime = update_context->m_AccumFrameTime * update_context->m_TimeScale;
+            }
+
+            const float time = collection->m_FixedAccumTime + update_context->m_DT; // Add the scaled time
+            const float fixed_frequency = update_context->m_FixedUpdateFrequency;
+            // If the proxy is slowed down, we want e.g. the physics to be slowed down as well
+            const float fixed_dt = (1.0f / (float)fixed_frequency) * update_context->m_TimeScale;
+            num_fixed_steps = (uint32_t)(time / fixed_dt);
+            // Store the remainder for the next frame
+            collection->m_FixedAccumTime = time - (num_fixed_steps * fixed_dt);
+
+            if (num_fixed_steps != 0)
+            {
+                fixed_update_context = dynamic_update_context;
+                fixed_update_context.m_DT = fixed_dt;
+            }
+        }
+
+        /* The overall function order is following:
```

Comment:
A lot of good notes.
However, a bunch ot spelling errors which makes it harder to read, but also find places in code which refers to these words.